Remove/Unexpose SVGDocument IDL

Ahmad-S792 opened the pull request:
#### 2330d8fd04640de7fc3f4e334498a07513141880
<pre>
Remove/Unexpose SVGDocument IDL

<a href="https://bugs.webkit.org/show_bug.cgi?id=269627">https://bugs.webkit.org/show_bug.cgi?id=269627</a>
<a href="https://rdar.apple.com/123121696">rdar://123121696</a>

Reviewed by Darin Adler.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

We already removed it in 178778@main but we had `LegacyWindowAlias` set
on `XMLDocument` interface but it is not on &apos;XMLDocument&apos; specification [1]:

[1] <a href="https://dom.spec.whatwg.org/#xmldocument">https://dom.spec.whatwg.org/#xmldocument</a>

This patch removes it and progress web platform test as well.

Credits to @karlcow for suggesting fix for `svg-tooltip.js`.

* Source/WebCore/dom/XMLDocument.idl:
* LayoutTests/svg/hittest/resources/svg-tooltip.js: Updated
* LayoutTests/imported/w3c/web-platform-tests/svg/historical-expected.txt:

Canonical link: <a href="https://commits.webkit.org/288135@main">https://commits.webkit.org/288135@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/efcb569607a7f143a0da07539d24ca42aef951f8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/81842 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1368 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35796 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86390 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/32838 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/83948 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1401 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9189 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63824 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21560 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/84912 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1028 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74478 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44110 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/927 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28653 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31291 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72289 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29260 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87826 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9082 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/6470 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72181 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9267 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70299 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71432 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17823 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15504 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14426 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/448 "") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9033 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/14565 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/8874 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12397 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10682 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

karlcow commented on LayoutTests/svg/hittest/resources/svg-tooltip.js:
```diff
@@ -11,7 +11,7 @@
 function canReturnTitle(element)
 {
-  if (!element || element.tagName == "svg" && element.ownerDocument instanceof SVGDocument)
+  if (!element || (element.tagName === "svg" && element.ownerDocument instanceof Document && element.namespaceURI === "http://www.w3.org/2000/svg"))
```
It might be interesting to figure out why there was a need for `&& element.ownerDocument instanceof SVGDocument` in the first place in the test.

The test was added in https://bugs.webkit.org/show_bug.cgi?id=139690 from 2014.

https://github.com/WebKit/WebKit/blob/a227ef91c9162e0519ccc17f99184bde84eac46e/LayoutTests/svg/hittest/resources/svg-tooltip.js#L12-L18

The test says:

```
// This code mimics the C++ code for calculating the tooltip text for an SVG element.
```

The current code to return the title inside SVG in the c++ code is 

https://searchfox.org/wubkat/rev/72adbb9f99534551a728acb270d4f111da5e9515/Source/WebCore/svg/SVGElement.cpp#820-829 

```cpp
String SVGElement::title() const
{
    // According to spec, for stand-alone SVG documents we should not return a title when
    // hovering over the rootmost SVG element (the first <title> element is the title of
    // the document, not a tooltip) so we instantly return.
    if (isOutermostSVGSVGElement() && document().topDocument().isSVGDocument())
        return String();
    RefPtr firstTitle = childrenOfType<SVGTitleElement>(*this).first();
    return firstTitle ? const_cast<SVGTitleElement&>(*firstTitle).innerText() : String();
}
```

So it tests if it is the `svg` root element AND if it is a standalone document (aka not inside an HTML document).

Another way to test this would be to test the `element.parentElement` It will be null in a standalone SVG for the `svg` element, while it would return another element with SVG embedded inside HTML.

karlcow commented on LayoutTests/svg/hittest/svg-embedded-tooltip-expected.txt:
```diff
@@ -1,7 +1,17 @@
+	actual:		'outermost svg'
+	expected:	''
+
+
```
The test is here
https://searchfox.org/wubkat/source/LayoutTests/svg/hittest/svg-embedded-tooltip.html

The test is failing because even for inline SVG, the namespace URI is 

```
> # testing with inline SVG
> $0.ownerDocument instanceof Document
< true
> $0.tagName
< "svg"
> $0.namespaceURI
< "http://www.w3.org/2000/svg"
> $0.parentElement
< <body>…</body>
```

testing the `parentElement` would be better.

karlcow commented on LayoutTests/svg/hittest/resources/svg-tooltip.js:
```diff
@@ -11,7 +11,7 @@
 function canReturnTitle(element)
 {
-  if (!element || element.tagName == "svg" && element.ownerDocument instanceof SVGDocument)
+  if (!element || (element.tagName === "svg" && element.ownerDocument instanceof Document && element.namespaceURI === "http://www.w3.org/2000/svg"))
```
```suggestion
  if (!element || (element.tagName === "svg" && element.ownerDocument instanceof XMLDocument))
```

karlcow replied:
Let me fix this above.

karlcow replied:
@Ahmad-S792 This will pass.

Ahmad-S792 replied:
@karlcow - Thanks for your help, it works. Now we can land this.

Ahmad-S792 committed:
Remove/Unexpose SVGDocument IDL

https://bugs.webkit.org/show_bug.cgi?id=269627
rdar://123121696

Reviewed by Darin Adler.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

We already removed it in 178778@main but we had `LegacyWindowAlias` set
on `XMLDocument` interface but it is not on 'XMLDocument' specification [1]:

[1] https://dom.spec.whatwg.org/#xmldocument

This patch removes it and progress web platform test as well.

Credits to @karlcow for suggesting fix for `svg-tooltip.js`.

* Source/WebCore/dom/XMLDocument.idl:
* LayoutTests/svg/hittest/resources/svg-tooltip.js: Updated
* LayoutTests/imported/w3c/web-platform-tests/svg/historical-expected.txt:

Canonical link: https://commits.webkit.org/288135@main
